### PR TITLE
Add materials for full Python 3.12 tutorial

### PR DESCRIPTION
**Pull request by gahjelle**
**Where to put new files:**

- New files should go into a top-level subfolder, named after the article slug. For example: `my-awesome-article`

**How to merge your changes:** 

1. [Make sure the CI code style tests all pass (+ run the automatic code formatter if necessary).](https://github.com/realpython/materials/blob/master/README.md)
2. Find an RP Team member on Slack and ask them to review & approve your PR.
3. Once the PR has one positive ("approved") review, GitHub lets you merge the PR.
4. 🎉

### CODE REVIEW

**Review by martin-martin — APPROVED** [review]
Looks good to me, one question about the commented-out code included below. :)

File: python-312/typing/generic_stack.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        self.stack: list[T] = []
+
+    def push(self, element: T) -> None:
+        self.stack.append(element)
+
+    def pop(self) -> T:
+        return self.stack.pop()
+
+
+# %% Python 3.12
```
[the remark this answers is not in the document]
Comment (gahjelle) [reply python-312/typing/generic_stack.py:17]: Hehe, the `# %%` syntax is used by some editors like Spyder and VS Code to divide source code into "cells". It doesn't really matter here, but the example looks more prominent in those editors that support this convention:

image

File: python-312/typing/generic_stack.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+# class Stack[T]:
+#     def __init__(self) -> None:
+#         self.stack: list[T] = []
+#
+#     def push(self, element: T) -> None:
+#         self.stack.append(element)
+#
+#     def pop(self) -> T:
+#         return self.stack.pop()
```
[the remark this answers is not in the document]
Comment (martin-martin) [reply python-312/typing/generic_stack.py:27]: Ah right, the linter!

**Review by KateFinegan — APPROVED** [review]
Your additions look good to me, @gahjelle! Please feel free to merge this PR once you're scheduled the tutorial.

**Pull request merged**